web_src/js/svg.js

```diff
@@ -38,18 +40,33 @@ export const svgs = {
 
 const parser = new DOMParser();
 const serializer = new XMLSerializer();
+const parsedSvgs = new Map();
 
-// retrieve a HTML string for given SVG icon name, size and additional classes
-export function svg(name, size = 16, className = '') {
+function getParsedSvg(name) {
+  if (parsedSvgs.has(name)) return parsedSvgs.get(name);
+  const root = parser.parseFromString(svgs[name], 'text/html');
+  const svgNode = root.querySelector('svg');
+  parsedSvgs.set(name, svgNode);
```

Comment:
A question about `parsedSvgs` cache:

Will the style conflict if there are both `svgNode('foo', 16)` and `svgNode('foo', 32)`? It seems that they use the same node object.

Reply:
`getParsedSvg` always returns default size (16), but `svgNode` will transform via `applyAttributes` if size differs. this cache only holds the default size, which is the common case.

Reply:
I mean if we do :

```
svg1 = svgNode('foo', 16)
insert(svg1)

svg2 = svgNode('foo', 32)
insert(svg2)

// actually, svg1 === svg2, is the same object from cache.
// so, will the first inserted svg becomes size=32?
```

Reply:
You're right, the reference stays the same on subsequent retrievals, I added a `cloneNode` to fix it and also some tests in https://github.com/go-gitea/gitea/pull/17638/commits/0b7374939bb0ac037d71e40ed5b261dcbe16c87b.